### DISCUSSION TITLE
How to associate ue_id of the RIC with the net_ns of the UE (srsRAN_4G)

### DISCUSSION THREAD

**Discussion by mas-oliveira** [srsran/oran-sc-ric#7] [General]
Hello,

I'm using format 5 to subscribe to the information of 3 UEs (srsRAN_4G with ZMQ) that I'm running. I'm able to connect them all and subscribe to each of them, but I can notice that if I start to generate traffic with the ue1 (related to ue1_zmq.conf), sometimes it is assigned to ue_id 0 on the controller, other times to ue_id 1, and even to ue_id 2. I think that, since I'm launching all the UEs at the same time (using the GNU Radio) they're randomly assigned to this ue_id in the controller.

I'm always using this command to launch the xApp:

> python3 kpm_mon_xapp.py --metrics=DRB.UEThpDl,DRB.UEThpUl,DRB.RlcPacketDropRateDl,DRB.RlcSduTransmittedVolumeDL,DRB.RlcSduTransmittedVolumeUL,DRB.PacketSuccessRateUlgNBUu,RRU.PrbAvailDl,RRU.PrbAvailUl,RRU.PrbTotDl,RRU.PrbTotUl --kpm_report_style=5 --ue_ids=0,1,2

I would like to ask if someone knows some way that I can get more info about the UE that's assigned in the controller as ue_id = {0,1,2}. I.e., if I can know in the controller which local UE is associated with each controller UE ID.

Thanks

**Comment by mas-oliveira** [comment]
It will be anything related with this in the next update?

**Reply by pgawlowicz** [reply]
I am not sure whether it is possible as there are no permanent UE IDs in RAN. Only the core network knows IMSI, but then generates a new UE IDs to talk with gnb.

**Comment by OniReimu** [comment]
> Hello,
> 
> I'm using format 5 to subscribe to the information of 3 UEs (srsRAN_4G with ZMQ) that I'm running. I'm able to connect them all and subscribe to each of them, but I can notice that if I start to generate traffic with the ue1 (related to ue1_zmq.conf), sometimes it is assigned to ue_id 0 on the controller, other times to ue_id 1, and even to ue_id 2. I think that, since I'm launching all the UEs at the same time (using the GNU Radio) they're randomly assigned to this ue_id in the controller.
> 
> I'm always using this command to launch the xApp:
> 
> > python3 kpm_mon_xapp.py --metrics=DRB.UEThpDl,DRB.UEThpUl,DRB.RlcPacketDropRateDl,DRB.RlcSduTransmittedVolumeDL,DRB.RlcSduTransmittedVolumeUL,DRB.PacketSuccessRateUlgNBUu,RRU.PrbAvailDl,RRU.PrbAvailUl,RRU.PrbTotDl,RRU.PrbTotUl --kpm_report_style=5 --ue_ids=0,1,2
> 
> I would like to ask if someone knows some way that I can get more info about the UE that's assigned in the controller as ue_id = {0,1,2}. I.e., if I can know in the controller which local UE is associated with each controller UE ID.
> 
> Thanks

Hi @mas-oliveira, @pgawlowicz  , just wondering to run multiple UEs, what extra configurations are needed? Configure `ue_zmq.conf`? Run `sudo ip netns add` to create more UEs? Update routing table? something else? Thanks.

**Reply by pgawlowicz** [reply]
@OniReimu please check this tutorial: https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html#multi-ue-emulation
